### PR TITLE
Fix: 7173 Returning Locale.getDefault

### LINKED ISSUE #7173: System Language "Traditional Chinese" selects "Simplified Chinese" as default language
###### Reproduction Steps

1. Set System Language to Traditional Chinese
2. Clear AnkiDroid Data
3. Open AnkiDroid

###### Expected Result

4. AnkiDroid is in Traditional Chinese

###### Actual Result

4. AnkiDroid is in Simplified Chinese

###### Research
- [x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] I have searched for similar existing issues here and on the user forum
- [x] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

### PR DESCRIPTION
**Pull request by kartikeysaran**
## Purpose / Description
Returning Locale.getDefault()

## Fixes
Fixes #7173 

## How Has This Been Tested?

This has been tested on a Real device

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/utils/LanguageUtil.java [hunk 1]
```diff
@@ -94,10 +94,10 @@ public static Locale getLocale(@Nullable String localeCode, @NonNull SharedPrefe
                 locale = new Locale(localeParts[0], localeParts[1]);
             } catch (ArrayIndexOutOfBoundsException e) {
                 Timber.w(e, "LanguageUtil::getLocale variant split fail, using code '%s' raw.", localeCode);
-                locale = new Locale(localeCode);
+                locale = Locale.getDefault();
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/utils/LanguageUtil.java:97]: This returns the locale of the system, not the locale of the app

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
I don't think this is correct

**Comment by david-allison-1** [comment]
Regardless, for something like this, a unit test displaying the problem would be fantastic

**Comment by kartikeysaran** [comment]
> I don't think this is correct

I had cleared the App cache and then reopened the app the app was in Traditional Chinese would recheck this once

**Comment by mikehardy** [comment]
going stale but it was a solid user complaint - would like to review + harvest the initial idea here if possible

**Pull request closed without merging**